src/graphql/queries.ts

```diff
@@ -132,3 +132,37 @@ export const getLPTokensQuery = () => {
   `
   return gql(queryString)
 }
+
+export const getBlocksQuery = (timestamps: Array<number>) => {
+  let queryString = 'query blocks {'
+  queryString += timestamps.map((timestamp) => {
+    return `t${timestamp}:blocks(first: 1, orderBy: timestamp, orderDirection: desc, where: { timestamp_gt: ${timestamp}, timestamp_lt: ${
+      timestamp + 600
+    } }) {
+      number
+    }`
+  })
+  queryString += '}'
+  return gql(queryString)
+}
+
+export const getPricesByBlockQuery = (tokenAddress: string, blocks: Array<any>) => {
+  let queryString = 'query blocks {'
```

Comment:
can you refactor the querytring to use string interpolation? like
`queryString = `query ${firstpart}, ${secondPart)`